core/src/com/unciv/logic/GameInfo.kt

```diff
@@ -260,16 +262,70 @@ class GameInfo {
                 changeBuildingName(cityConstructions, "Machu Pichu", "Machu Picchu")
                 // As of 2.16.1, changed Colloseum to Colosseum
                 changeBuildingName(cityConstructions, "Colloseum", "Colosseum")
+
+                // As of 3.4.9 cities have a unique identifier
+                if (cityInfo.id == "") cityInfo.id = UUID.randomUUID().toString()
             }
 
             // This doesn't HAVE to go here, but why not.
             // As of version 3.1.3, trade offers of "Declare war on X" and "Introduction to X" were changed to X,
             //   with the extra text being added only on UI display (solved a couple of problems).
-            for(trade in civInfo.tradeRequests.map { it.trade })
-                for(offer in trade.theirOffers.union(trade.ourOffers)){
+            for(trade in civInfo.tradeRequests.map { it.trade }) {
+                for (offer in trade.theirOffers.union(trade.ourOffers)) {
                     offer.name = offer.name.removePrefix("Declare war on ")
                     offer.name = offer.name.removePrefix("Introduction to ")
                 }
+            }
+
+            // As of 3.4.9 cities are referenced by id, not by name
+            // So try to update every tradeRequest (if there are no conflicting names)
```

Comment:
not for this PR, but for future refatoring
looks like this is similar to SQL migrations and better to create main func `runMigrations` and split each version in own private func and call them by order in main func

Reply:
Totally agree, this is the `GameSanitizer` I was mentioning in the PR message.

Reply:
Finally I understand what is 'gameSanitizer' lol